core/src/main/java/org/mule/api/security/tls/TlsConfiguration.java

```diff
@@ -305,37 +334,192 @@ private void initTrustManagerFactory() throws CreateException
     {
         if (null != trustStoreName)
         {
-            trustStorePassword = null == trustStorePassword ? "" : trustStorePassword;
+            KeyStore trustStore = createTrustStore();
 
-            KeyStore trustStore;
             try
             {
-                trustStore = KeyStore.getInstance(trustStoreType);
-                InputStream is = IOUtils.getResourceAsStream(trustStoreName, getClass());
-                if (null == is)
+                trustManagerFactory = TrustManagerFactory.getInstance(trustManagerAlgorithm);
+
+                Boolean revocationEnabled = getRevocationCheck() != null;
+                ManagerFactoryParameters tmfParams = null;
+
+                // Revocation checking is only supported for PKIX algorithm
+                if (revocationEnabled && "PKIX".equalsIgnoreCase(getTrustManagerAlgorithm()))
                 {
-                    throw new FileNotFoundException(
-                            "Failed to load truststore from classpath or local file: " + trustStoreName);
+                    if (getRevocationCheck().getStandardRevocationCheck() != null) {
```

Comment:
Considering that you are doing the same thing for each revocation type it looks like you are missing an abstraction for these types. I think RevocationCheck should contain just one such abstraction (or even that it should be the abstraction) which returns the params when asked for them. That will make the code much simpler and `configFor` methods could live in the actual classes that define each one.